site/content/docs/developing/service-discovery.en.md

```diff
@@ -33,6 +33,12 @@ The important part is that our `front-end` service is making a request to our `a
 endpoint := fmt.Sprintf("http://api.%s/some-request", os.Getenv("COPILOT_SERVICE_DISCOVERY_ENDPOINT"))
 ```
 
-`COPILOT_SERVICE_DISCOVERY_ENDPOINT` is a special environment variable that the Copilot CLI sets for you when it creates your service. It's of the format _{app name}.local_ - so in this case in our _kudos_ app, the request would be to `http://api.kudos.local/some-request`. Since our _api_ service is running on port 80, we're not specifying the port in the URL. However, if it was running on another port, say 8080, we'd need to include the port in the request, as well `http://api.kudos.local:8080/some-request`.
+`COPILOT_SERVICE_DISCOVERY_ENDPOINT` is a special environment variable that the Copilot CLI sets for you when it creates your service. It's of the format _{env name}.{app name}.local_ - so in this case in our _kudos_ app, when deployed in the _test_ environment, the request would be to `http://api.test.kudos.local/some-request`. Since our _api_ service is running on port 80, we're not specifying the port in the URL. However, if it was running on another port, say 8080, we'd need to include the port in the request, as well `http://api.kudos.local:8080/some-request`.
 
-When our front-end makes this request, the endpoint `api.kudos.local` resolves to a private IP address and is routed privately within your VPC. 
+When our front-end makes this request, the endpoint `api.test.kudos.local` resolves to a private IP address and is routed privately within your VPC. 
+
+## Legacy Environments and Service Discovery
+
+Prior to Copilot v1.8.0 and environment version 1.5.0, the service discovery namespace used the format _{app name}.local_, without including the environment. This limitation made it impossible to deploy multiple environments in the same VPC. Any environments created with Copilot v1.8.0 and newer can share a VPC with any other environment.
+
+When your environments are upgraded, Copilot will honor the service discovery namespace that the environment was created with. That means that the endpoint your services are reachable at will not change. Any new environments created with Copilot v1.8.0 and above will use the _{env name}.{app name}.local_ format for service discovery, and can share VPCs with older environments. 
```

Comment:
Can we create an issue and pin that in our repository so that folks are aware of this change.  
Ideally if they use only the env variable then we are good. However, if folks have `app.local` hardcoded in their code and deploy to a new env then their code will fail mysteriously. (Which I think is ok since we provide the env variable)